### PR TITLE
snapshot: introducing snapshot pkg

### PR DESCRIPTION
**Pull request by ehnuje**
## Proposed changes

- This is the first PR to bring snapshot package from Geth to Klaytn
- As it is not created anywhere, it does not affect other package or Klaytn routine for now
- StateDB and BlockChain related PR will be uploaded later and then snapshot pkg can be used

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: snapshot/account.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 The go-ethereum Authors
```
Comment (yoomee1313) [inline snapshot/account.go:1]: Klaytn copyright is missing at some files!

File: snapshot/conversion.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+		} else {
+			stats.progress(0, processed, account, last)
+		}
+	}
+	result := stop(true)
+	return result, nil
+}
+
+// stdGenerate is a very basic hexary trie builder which uses the same Trie
+// as the rest of geth, with no enhancements or optimizations
```
Comment (yoomee1313) [inline snapshot/conversion.go:268]:
```suggestion
// as the rest of the code, with no enhancements or optimizations
```
what about this?

File: snapshot/wipe.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	}()
+	return wiper
+}
+
+// wipeContent iterates over the entire key-value database and deletes all the
+// data associated with the snapshot (accounts, storage), but not the root hash
+// as the wiper is meant to run on a background thread but the root needs to be
+// removed in sync to avoid data races. After all is done, the snapshot range of
+// the database is compacted to free up unused data blocks.
+func wipeContent(db database.DBManager) error {
```
Comment (winnie-byun) [inline snapshot/wipe.go:54]: When we are using non-singleDB, we could just remove the DB?

File: storage/database/schema.go [hunk 4]
```diff
@@ -50,6 +50,22 @@ var (
 
 	snapshotKeyPrefix = []byte("snapshot")
```
Comment (winnie-byun) [inline storage/database/schema.go:51]:
```suggestion
	// snapshotKeyPrefix tracks Istanbul snapshot.
	snapshotKeyPrefix = []byte("snapshot")
```
The two snapshot is confusing a little.

Comment (winnie-byun) [reply storage/database/schema.go:51]:
```suggestion
	IstanbulSnapshotKeyPrefix = []byte("snapshot")
```
What do you think of putting `istanbul`?

**Comment by KimKyungup** [comment]
When we implement the snapshot, this will be reviewed and merged.

**Comment by ehnuje** [comment]
As we already ported snapshot, let's close this PR.

**Pull request closed without merging**